File: Algorithms/p300_Longest_Increasing_Subsequence_1.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        if(nums.empty()) return 0;
        int n = nums.size();
        vector<int> table(n,1);
        for(int i=0;i<n;++i){
            for(int j=0;j<i;++j){
                if(nums[i]>nums[j]) table[i] = max(table[i],table[j]+1);
            }
        }
        return *max_element(table.begin(),table.end());
    }
};

File: Algorithms/p300_Longest_Increasing_Subsequence_1.py
class Solution(object):
    def lengthOfLIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        n = len(nums)
        table = [1]*n
        for i in range(n):
            for j in range(i):
                if nums[i]>nums[j]:
                    table[i] = max(table[i],table[j]+1)
        return max(table)
